runtime/tools/python/ttrt/common/util.py

```diff
@@ -616,16 +621,23 @@ def __init__(self, index, program):
             self.program = program
             self.input_tensors = []
             self.output_tensors = []
+            self.golden_map = GoldenMap()
 
         def populate_inputs(self, init_fn):
-            for i in self.program["inputs"]:
-                torch_tensor = init_fn(
-                    i["desc"]["shape"],
-                    dtype=Binary.Program.from_data_type(
-                        i["desc"]["layout"]["memory_desc"]["data_type"]
-                    ),
-                )
-                self.input_tensors.append(torch_tensor)
+            inputs = self.golden_map.get_inputs()
```

Comment:
It's unclear if we need a whole `GoldenMap` data structure.  I think it might be cleaner to detect if there are goldens present in the flatbuffer above this method and then just pass in an `init_fn` that knows how to look them up from the Binary, or just pass in a list of pre-initialized inputs.

Reply:
removed old code that's not needed and cleaned up populating inputs